### DISCUSSION TITLE
APT repository key file

### DISCUSSION THREAD

**Discussion by joos2k** [authelia/authelia#5272] [Q&A] [answered]
In Debian apt-key is deprecated and shouldn't be used anymore for security reasons.

If you try to use the Debian recommended method the signing key for the repository could not be found by apt.
Problem is: In the signing.asc file the line endings are not LF (unix). You need to convert those endings (for example with dos2unix) and then it works.

The recommended method is this:
1. put the key file in some directory (I use /etc/apt/keyrings, but others are fine too - just don't use /etc/apt/trusted.gpg.d)
2. add a reference to the key file in the deb-line with "deb [signed-by=/etc/apt/keyrings/authelia.key] https://apt.auth..."

Details are here: https://wiki.debian.org/DebianRepository/UseThirdParty

I think you should change the line endings in the key file on your side, so new repository users don't run into this issue. :-)

**Comment by nightah** [comment, marked as answer]
Thanks for the heads up @joos2k, we'll see what we can do about both changing the line endings for the signing key and also updating the instructions on how to properly sign repositories.

**Reply by nightah** [reply]
@joos2k we've updated the signing key to have Unix line endings could you perhaps try again and confirm that this is resolved?

We'll also be updating the instructions on "How to use this repo" to include the newer/recommended method.

**Reply by joos2k** [reply]
@nightah I tried it and now it works as it should. Thanks. :-)

Here is my suggestion for the instructions update:
```
curl -s https://apt.authelia.com/organization/signing.asc | sudo tee /usr/share/keyrings/authelia.asc > /dev/null

echo "deb [signed-by=/usr/share/keyrings/authelia.asc] https://apt.authelia.com/stable/debian/debian/ all main" | sudo tee /etc/apt/sources.list.d/authelia-stable-debian.list
```
